src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -380,7 +380,7 @@ private static bool SymbolsAreCompatibleCore(ISymbol symbol, ISymbol newSymbol,
                     var type = methodSymbol.ContainingType;
                     var newType = newMethodSymbol.ContainingType;
                     if ((type != null && type.IsEnumType() &&
-                         type.EnumUnderlyingType != null && type.EnumUnderlyingType.SpecialType == newType.SpecialType) ||
+                         type.EnumUnderlyingType != null && type.EnumUnderlyingType.SpecialType == newType?.SpecialType) ||
```

Comment:
The second part of the overarching `&&` condition indicates that `newType == null` is a real possibility here.  Hence I used an `?.` to account for that.  I'm not particularly familiar with this code hence there may be a better fix here.